main.py

```diff
@@ -0,0 +1,54 @@
+# Import the requests library
+import requests
+import random
+
+# Define the URL for the random word API
+url = 'https://random-word-api.herokuapp.com/word'
+
+
+# Define an empty list, this list will be filled with the provided
+# results and used to pick a random name as final suggestion
+generated_names = []
+
+
+# Define a function called "results"
+# that takes a URL, a number, and a name as inputs
+def results(url, num, name):
+    # This loop to repeat the creation and display of
+    # names provided a number of times provided by the user
+    for i in range(num):
+        # API call and cleaning of the result
+        api_call = requests.get(url)
+        nickname = api_call.text.replace(
+            '[', '').replace(']', '').replace('"', '')
+        # Show the final results and add it to the
+        # generated_names list
+        madeup_name = (f"\n{name.title()} {nickname.title()}")
+        print(madeup_name)
+        generated_names.append(madeup_name)
+
+
+# This asks the user for a name as starting point
+# and check is an acceptabe input
+while True:
+    name = str(input("Please provide a first name:"))
+    if len(name) >= 2 and isinstance(name, str):
+        break
+    else:
+        print("don't be a fool a give me a real name!")

```

Comment:
It would be great if you could recheck the minor grammatical errors and punctuations!! Just a suggestion for this outstanding code.